src/openforms/submissions/api/serializers.py

```diff
@@ -246,6 +247,12 @@ def to_representation(self, instance):
             instance.submission.data,
             **self.context,
         )
+
+        # Add to each component the custom errors in the current locale
+        new_configuration = get_translated_custom_error_messages(
+            new_configuration, instance.submission
+        )
```

Comment:
wouldn't it make more sense to hook this into `openforms.formio.service.get_dynamic_configuration`?

This sets up the formio configuration wrapper once (which avoids many `iter_components` calls for performance reasons) and allows you to build on that datastructure. `get_dynamic_configuration` is called by `evaluate_form_logic`.